api/v1alpha1/addon_types.go

```diff
@@ -257,4 +257,29 @@ type ServiceLoadBalancer struct {
 	// +kubebuilder:validation:Enum=MetalLB
 	// +kubebuilder:validation:Required
 	Provider string `json:"provider"`
+
+	// Configuration for the chosen ServiceLoadBalancer provider.
+	// +kubebuilder:validation:Optional
+	Configuration *ServiceLoadBalancerConfiguration `json:"configuration,omitempty"`
+}
+
+type ServiceLoadBalancerConfiguration struct {
+	// AddressRanges is a list of IPv4 or IPv6 address ranges from which
+	// the provider can choose a load balancer address.
+	// +kubebuilder:validation:MinItems=1
```

Comment:
Please add  required too - I've been advised to do this upstream even though it is a default as there have been reports of this not being consistently handled in controller-gen.
```suggestion
	// +kubebuilder:validation:Required
	// +kubebuilder:validation:MinItems=1
```

Reply:
Done in #778